src/hooks/search/index.ts

```diff
@@ -93,126 +99,132 @@ const useSearch = (
         };
     }, []);
 
+    const getFiltersFromQueryStr = () => {
+        const {
+            l1Items,
+            contentTypeItems,
+            languageItems,
+            technologyItems,
+            contributedByItems,
+            expertiseLevelItems,
+        } = filterItems as SearchFilterItems;
+        const {
+            product,
+            language,
+            technology,
+            contributedBy,
+            contentType,
+            expertiseLevel,
+        } = router.query;
+
+        let allNewFilters: FilterItem[] = [];
+        if (product) {
```

Comment:
might be a little bit of a pain to do, but this `if` and the `if` on line 147 look pretty similar (unless im missing something), they could possibly be extracted to a function and then just passed the relevant arrays, etc. 

Same with the if statements from 174 through 212